server_extensions_extra/src/main/java/com/generalbytes/batm/server/extensions/extra/aeternity/AeternityPaymentSupport.java

```diff
@@ -0,0 +1,130 @@
+    @Override
+    public long getMaximumWaitForPossibleRefundInMillis() {
+        return MAXIMUM_WAIT_FOR_POSSIBLE_REFUND_MILLIS;
+    }
+
+    @Override
+    public BigDecimal getTolerance() {
+        return TOLERANCE;
+    }
+
+    @Override
+    public BigDecimal getMinimumNetworkFee(RPCClient client) {
+        return TOLERANCE;
+    }
+
+    @Override
+    public ICryptoAddressValidator getAddressValidator() {
+        return addressValidator;
+    }
+
+    
+  
+    public BigDecimal calculateTxFee(String senderAddress, String destinationAddress, BigDecimal amount,  RPCClient client) {
+    	return ((AeternityRPCClient)client).calculateTxFee(senderAddress, destinationAddress, amount, client);
+    }
+//    public static void main(String[] args) {
+//        //You need to have node running: i.e.:  bitcoind -rpcuser=rpcuser -rpcpassword=rpcpassword -rpcport=8332
+//
+//        BitcoinCashRPCWallet wallet = new BitcoinCashRPCWallet("http://rpcuser:rpcpassword@localhost:8332", "");
+//        BitcoinCashPaymentSupport ps = new BitcoinCashPaymentSupport();
+//        ps.init(null);
+//        PRS spec = new PRS(
+//            ps.getCurrency(),
+//            "Just a test",
+//            60 * 15, //15 min
+//            3,
+//            false,
+//            false,
+//            new BigDecimal("7"),
+//            new BigDecimal("10"),
+//            wallet
+//        );
+//        spec.addOutput("qpqkqq2uy6v044yjqsec0cprunecwcf9dqtc5ler86", new BigDecimal("0.0017"));
+//
+//        PaymentRequest pr = ps.createPaymentRequest(spec);
+//        System.out.println(pr);
+//        pr.setListener(new IPaymentRequestListener() {
+//            @Override
+//            public void stateChanged(PaymentRequest request, int previousState, int newState) {
+//                System.out.println("stateChanged = " + request + " previousState: " + previousState + " newState: " + newState);
+//            }
+//
+//            @Override
+//            public void numberOfConfirmationsChanged(PaymentRequest request, int numberOfConfirmations, Direction direction) {
+//                System.out.println("numberOfConfirmationsChanged = " + request + " numberOfConfirmations: " + numberOfConfirmations + " direction: " + direction);
+//            }
+//
+//            @Override
+//            public void refundSent(PaymentRequest request, String toAddress, String cryptoCurrency, BigDecimal amount) {
+//                System.out.println("refundSent = " + request + " toAddress: " + toAddress +" cryptoCurrency: " + cryptoCurrency + " " + amount);
+//            }
+//        });
+//        System.out.println("Waiting for transfer");
+//        try {
+//            Thread.sleep(20 * 60 * 1000);
+//        } catch (InterruptedException e) {
+//            e.printStackTrace();
+//        }
+//    }
+
+
+    @Override
+    public String getSigHashType() {//This is BitcoinCash specific. No need to override this method for other currencies
+        return "ALL|FORKID";
```

Comment:
Is this really valid for AE?

Reply:
nope, will remove this override for Aeternity